src/main/java/com/iexec/worker/pubsub/StompClient.java

```diff
@@ -0,0 +1,169 @@
+        this.sessionRequestQueue.offer(new SessionRequestEvent());
+    }
+
+    /**
+     * Listen to new session request events and refresh the websocket
+     * connection by establishing a new STOMP session. Only one of 
+     * the received requests in a fixed time interval
+     * (SESSION_REFRESH_DELAY) will be processed. We use @Scheduled
+     * to start the watcher asynchronously with an initial delay and
+     * restart it in case a problem occurs.
+     * 
+     * @throws InterruptedException
+     */
+    @Scheduled(fixedDelay = 1000)
+    private void listenToSessionRequestEventsInTheQueue() throws InterruptedException {
+        while (true) {
+            // get the first request event or wait until available
+            this.sessionRequestQueue.take();
+            // wait some time for the wave of request events coming
+            // from possibly different threads to finish
+            TimeUnit.SECONDS.sleep(SESSION_REFRESH_DELAY);
+            log.info("Creating new STOMP session");
+            // purge redundant request events
+            this.sessionRequestQueue.clear();
+            // Only one attempt should pass through
+            log.debug("Sending new STOMP connection request");
+            this.stompClient.connect(webSocketServerUrl, new SessionHandler());
+        }
+    }
+
+    /**
+     * Provide callbacks to handle STOMP session establishment or
+     * failure.
+     */
+    private class SessionHandler extends StompSessionHandlerAdapter {
+
+        @Override
+        public void afterConnected(StompSession session, StompHeaders connectedHeaders) {
+            log.info("Connected to STOMP session [session: {}, isConnected: {}]",
+                    session.getSessionId(), session.isConnected());
+            StompClient.this.session = session;
+            // notify subscribers
+            eventPublisher.publishEvent(new SessionCreatedEvent());
+        }
+
+        @Override
+        public void handleException(StompSession session, @Nullable StompCommand command,
+                                    StompHeaders headers, byte[] payload, Throwable exception) {
+            SimpMessageType messageType = command != null ? command.getMessageType() : null;
+            log.error("STOMP error [session: {}, isConnected: {}, command: {}, exception: {}]",
+                    session.getSessionId(), session.isConnected(), messageType, exception.getMessage());
+        }
+
+        @Override
+        public void handleTransportError(StompSession session, Throwable exception) {
+            log.error("STOMP transport error [session: {}, isConnected: {}, exception: {}]",
+                    session.getSessionId(), session.isConnected(), exception.getMessage());
+            requestNewSession();
+        }
+    }
+
+    /**
+     * Request a new STOMP session by adding
+     * this event to the queue.
+     */
+    @NoArgsConstructor
+    private class SessionRequestEvent {
```

Comment:
Probably move this to a single file (even if it's empty)

Reply:
That class is only used inside the `StompClient` class. It shouldn't be used outside that's why I left it as an inner class.

Reply:
Or do you mean this one 
```
    /**
     * Publish this event when a new session
     * is created to notify subscribers.
     */
    @NoArgsConstructor
    public class SessionCreatedEvent {
        
    }
```